chrome/content/zotero/bindings/itembox.xml

```diff
@@ -1068,9 +1069,14 @@
 						fields.fieldMode = fieldMode;
 						this.modifyCreator(index, fields);
 						if (this.saveOnEdit) {
-							this.item.saveTx();
+							let activeField = this._dynamicFields.querySelector('textbox');
+							if (activeField !== null && activeField !== firstName && activeField !== lastName) {
+								await this.blurOpenField();	
+							}
+							await this.item.saveTx();
```

Comment:
I think this can be in an `else`, since if a save is necessary it would've been triggered by blurring. (I know we have `blurOpenField` followed by `save` elsewhere, but in those cases we don't necessarily check whether a field is open. Here we know that a field is open.)

Reply:
Makes sense, thanks! This also means this function doesn't need to be `async`. I guess it works either way but I've changed it back to sync to keep the changes to minimum.